### PR TITLE
[CIS-697] Compose Avatar Views

### PR DESCRIPTION
**Pull request by nuno-vieira**
# In this PR

Uses composition instead of inheritance to create different avatar views types.

### `ChatAvatarView`
Component responsible to render the avatar image.

###  `ChatOnlineIndicatorView`
Component responsible to display an online indicator

### `ChatPresenceAvatarView`
- `ChatAvatarView`
- `ChatOnlineIndicatorView`

### `ChatUserAvatarView`
- `content: ChatUser`
- `ChatPresenceAvatarView`

### `ChatChannelAvatarView`
- `content: ChatChannel`
- `ChatPresenceAvatarView`

### CODE REVIEW

File: Sources/StreamChatUI/CommonViews/AvatarView/ChatOnlineAvatarView/ChatOnlineAvatarView.swift [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import UIKit
+
+/// A view that shows a user avatar including an indicator of the user presence (online/offline).
+public typealias ChatOnlineAvatarView = _ChatOnlineAvatarView<NoExtraData>
+
+/// A view that shows a user avatar including an indicator of the user presence (online/offline).
+open class _ChatOnlineAvatarView<ExtraData: ExtraDataTypes>: _View, UIConfigProvider {
+
+    /// A view that shows the avatar image
+    open private(set) lazy var avatarView: ChatAvatarView = ChatAvatarView()
```
Comment (VojtaStavik) [inline Sources/StreamChatUI/CommonViews/AvatarView/ChatOnlineAvatarView/ChatOnlineAvatarView.swift:15]: this should come from UIConfig

File: Sources/StreamChatUI/CommonViews/AvatarView/ChatOnlineAvatarView/ChatOnlineAvatarView.swift [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+//
+
+import StreamChat
+import UIKit
+
+/// A view that shows a user avatar including an indicator of the user presence (online/offline).
+public typealias ChatOnlineAvatarView = _ChatOnlineAvatarView<NoExtraData>
+
+/// A view that shows a user avatar including an indicator of the user presence (online/offline).
+open class _ChatOnlineAvatarView<ExtraData: ExtraDataTypes>: _View, UIConfigProvider {
```
Comment (VojtaStavik) [inline Sources/StreamChatUI/CommonViews/AvatarView/ChatOnlineAvatarView/ChatOnlineAvatarView.swift:12]: `ChatPresenceAvatarView` maybe?

File: Sources/StreamChatUI/CommonViews/AvatarView/ChatUserAvatarView.swift [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        didSet { updateContentIfNeeded() }
+    }
+
+    override open func setUpLayout() {
+        super.setUpLayout()
+        embed(onlineAvatarView)
+    }
+
+    override open func updateContent() {
+        guard let user = content else { return }
```
Comment (VojtaStavik) [inline Sources/StreamChatUI/CommonViews/AvatarView/ChatUserAvatarView.swift:30]: I don't think we need this check. We need to reset the UI when the content is `nil`

**Review by VojtaStavik — APPROVED** [review]
Very nice 👍

File: Sources/StreamChatUI/CommonViews/AvatarView/ChatUserAvatarView.swift (hunk 3 above)
Comment (nuno-vieira) [reply Sources/StreamChatUI/CommonViews/AvatarView/ChatUserAvatarView.swift:30]: Yeah good point 👍

File: Sources/StreamChatUI/CommonViews/AvatarView/ChatOnlineAvatarView/ChatOnlineAvatarView.swift (hunk 2 above)
Comment (nuno-vieira) [reply Sources/StreamChatUI/CommonViews/AvatarView/ChatOnlineAvatarView/ChatOnlineAvatarView.swift:12]: Yes I like that name better, thanks 👌

File: Sources/StreamChatUI/CommonViews/AvatarView/ChatOnlineAvatarView/ChatOnlineAvatarView.swift (hunk 1 above)
Comment (nuno-vieira) [reply Sources/StreamChatUI/CommonViews/AvatarView/ChatOnlineAvatarView/ChatOnlineAvatarView.swift:15]: Ah yes, nice catch 👍

Comment (nuno-vieira) [reply Sources/StreamChatUI/CommonViews/AvatarView/ChatOnlineAvatarView/ChatOnlineAvatarView.swift]: ✅

File: Sources/StreamChatUI/CommonViews/AvatarView/ChatOnlineAvatarView/ChatOnlineAvatarView.swift (hunk 2 above)
Comment (nuno-vieira) [reply Sources/StreamChatUI/CommonViews/AvatarView/ChatOnlineAvatarView/ChatOnlineAvatarView.swift]: ✅

File: Sources/StreamChatUI/CommonViews/AvatarView/ChatUserAvatarView.swift (hunk 3 above)
Comment (nuno-vieira) [reply Sources/StreamChatUI/CommonViews/AvatarView/ChatUserAvatarView.swift]: ✅

**Pull request merged**